staging/src/k8s.io/client-go/rest/request.go

```diff
@@ -938,7 +968,7 @@ func (r *Request) request(ctx context.Context, fn func(*http.Request, *http.Resp
 
 	client := r.c.Client
 	if client == nil {
-		client = http.DefaultClient
+		client = HttpClientResolverMetricsWrapper(http.DefaultClient)
```

Comment:
can we use an interface instead? this will allow more composability, also we can write better unit tests for this
```
type ClientDecorator interface {
    Decorate(*http.Client) *http.Client
}
```

we can compose it in `Request`
```
type Request struct {
     ClientDecorator ClientDecorator
}
```